src/librustc_mir/borrow_check.rs

```diff
@@ -449,51 +449,56 @@ impl<'c, 'b, 'a: 'b+'c, 'gcx, 'tcx: 'a> MirBorrowckCtxt<'c, 'b, 'a, 'gcx, 'tcx>
         // Check permissions
         self.check_access_permissions(lvalue_span, rw);
 
-        self.each_borrow_involving_path(
-            context, (sd, lvalue_span.0), flow_state, |this, _index, borrow, common_prefix| {
-                match (rw, borrow.kind) {
-                    (Read(_), BorrowKind::Shared) => {
-                        Control::Continue
-                    }
-                    (Read(kind), BorrowKind::Unique) |
-                    (Read(kind), BorrowKind::Mut) => {
-                        match kind {
-                            ReadKind::Copy =>
-                                this.report_use_while_mutably_borrowed(
-                                    context, lvalue_span, borrow),
-                            ReadKind::Borrow(bk) => {
-                                let end_issued_loan_span =
-                                    flow_state.borrows.base_results.operator().opt_region_end_span(
-                                        &borrow.region);
-                                this.report_conflicting_borrow(
-                                    context, common_prefix, lvalue_span, bk,
-                                    &borrow, end_issued_loan_span)
+        // Ignore all borrows rooted in statics
+        if let Lvalue::Local(_) = *self.prefixes(lvalue_span.0, PrefixSet::All).last().unwrap() {
```

Comment:
this `if let` makes me nervous -- if we add more lvalues in the future, it might silently squash them. I'd rather see an exhaustive `match`.

Reply:
Alternatively, something like:

```rust
if let Lvalue::Static(_) = ... {
    // Skip checks on statics. See #123.
} else {
    ...
}
```